lib/https.c

```diff
@@ -349,6 +350,38 @@ _establish_connection(struct https_request * const req,
 #endif /* HAVE_GETADDRINFO */
 }
 
+/* Provide implementations for HMAC_CTX_new and HMAC_CTX_free when
+ * building for OpenSSL versions older than 1.1.0
+ */
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+static void EVP_MD_CTX_free(EVP_MD_CTX *ctx)
+{
+    if (ctx != NULL) {
+        EVP_MD_CTX_cleanup(ctx);
```

Comment:
Oh. Actually, `HMAC_CTX_cleanup()` already does that for us, too:

```
void HMAC_CTX_cleanup(HMAC_CTX *ctx)
{
#ifdef OPENSSL_FIPS
    if (FIPS_mode() && !ctx->i_ctx.engine) {
        FIPS_hmac_ctx_cleanup(ctx);
        return;
    }
#endif
    EVP_MD_CTX_cleanup(&ctx->i_ctx);
    EVP_MD_CTX_cleanup(&ctx->o_ctx);
    EVP_MD_CTX_cleanup(&ctx->md_ctx);
    OPENSSL_cleanse(ctx, sizeof *ctx);
}
```

When in doubt, I tend to look at the compatibility shims in https://github.com/pyca/cryptography/; in this case https://github.com/pyca/cryptography/blob/master/src/_cffi_src/openssl/hmac.py. This is actually code that people run in production, rather than some apparently quite half-baked and unreviewed documentation :(